src/main/java/Main.java

```diff
@@ -1,27 +1,37 @@
-import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
+import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.util.*;
 
-public class Main {
+
+public class Main extends JsonDeserializer<Map<String, Results>> {
+
+
     public static void main(String[] args) throws IOException {
```

Comment:
Create new class for mapping, usage will be like:
`Student[] students = MyNameMapper.mapToObj("src/test/example.json", Student[].class)`

Also add method of mapping objects into files (you can ask details in messenger)
Finally show usage of your class in main method, remove all other